.github/workflows/test.yml

```diff
@@ -52,7 +49,147 @@ jobs:
         uses: actions/checkout@v3
       - name: Run unit tests
         run: make test
-      - name: Build testing
-        run: make build
       - name: Upload coverage reports to Codecov
         uses: codecov/codecov-action@v3
+      - name: Build bianry for E2E testing
+        run: |
+          ./scripts/build.sh v0.0.1 linux-x64
+          ./scripts/build.sh v0.0.1 win-x64
+          ./scripts/build.sh v0.0.1 osx-x64
+      - name: Upload Linux artifact
+        if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push'
+        uses: actions/upload-artifact@v2
+        with:
+          name: linux-amd64-binary
+          path: ./bin/artifacts/notation-azure-kv_0.0.1_linux_amd64.tar.gz
+          retention-days: 1
+      - name: Upload macOS artifact
+        if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push'
+        uses: actions/upload-artifact@v2
+        with:
+          name: darwin-amd64-binary
+          path: ./bin/artifacts/notation-azure-kv_0.0.1_darwin_amd64.tar.gz
+          retention-days: 1
+      - name: Upload Windows artifact
+        if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push'
+        uses: actions/upload-artifact@v2
+        with:
+          name: win-amd64-binary
+          path: ./bin/artifacts/notation-azure-kv_0.0.1_windows_amd64.zip
+          retention-days: 1
+  e2e-linux:
+    name: E2E testing on Linux
+    runs-on: ubuntu-latest
+    needs: test
+    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push'
+    steps:
+      - name: Check out code into the project directory
+        uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: linux-amd64-binary
+          path: ./bin/artifacts
+      - name: Run download server locally
+        run: |
+          nohup python3 ./test/e2e/download-server.py &
```

Comment:
Is it possible to use `python3 -m http.server`?

Reply:
Awsome! Updated.